zokrates_core/src/semantics.rs

```diff
@@ -2461,10 +2461,24 @@ impl<'ast, T: Field> Checker<'ast, T> {
                     }),
                 }
             }
-            Expression::Conditional(box condition, box consequence, box alternative, kind) => {
-                let condition_checked = self.check_expression(condition, module_id, types)?;
-                let consequence_checked = self.check_expression(consequence, module_id, types)?;
-                let alternative_checked = self.check_expression(alternative, module_id, types)?;
+            Expression::Conditional(box conditional) => {
+                let condition_checked =
+                    self.check_expression(*conditional.condition, module_id, types)?;
+
+                if !conditional.consequence_statements.is_empty()
+                    || !conditional.alternative_statements.is_empty()
+                {
+                    return Err(ErrorInner {
+                        pos: Some(pos),
+                        message: "statements in the if-else block are not yet supported"
```

Comment:
@Schaeff not sure how to cleanly write this, any better ideas?

Reply:
```suggestion
                        message: "Statements are not supported in conditional branches"
```

maybe like this?